src/main/java/com/minecolonies/entity/ai/minimal/EntityAIGateInteract.java

```diff
@@ -136,13 +137,14 @@ private boolean checkFenceGate(@NotNull PathEntity pathentity)
      */
     private BlockFenceGate getBlockFence(@NotNull BlockPos pos)
     {
-        Block block = this.theEntity.worldObj.getBlockState(pos).getBlock();
-        if (!(block instanceof BlockFenceGate && block.getMaterial() == Material.wood))
+        IBlockState blockState = this.theEntity.worldObj.getBlockState(pos);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'blockState' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)